src/cpp_vector.rs

```diff
@@ -0,0 +1,100 @@
+#[repr(C)]
+#[derive(Debug)]
+pub struct CppVector<T> {
+    start: *mut T,
+    end: *mut T,
+    eos: *mut T
+}
+
+impl<T> CppVector<T> {
+    pub fn new(len: usize, cap: usize) -> Self {
+        unsafe {
+            assert!(cap >= len);
```

Comment:
I'd recommend either
* marking new as track_caller and changing this to a panic with a good message for what went wrong or
* having new return a `Result<Self, InvalidCapacity>` with just `struct InvalidCapacity;`

the reasoning is think about how this will act when needed to be debugged: since it's an assert, all the person calling this incorrectly will get is a vague message about cap >= len and a useless line number (since it points to this file, not the callee). If you use one of the above it will ensure the line number of the error lines up with the code calling it + make it clear as to the cause.